pyfstat/make_sfts.py

```diff
@@ -129,10 +117,78 @@ def basic_setup(self):
             )
             for ind, dets in enumerate(IFOs)
         ]
+        self.IFOs = ",".join(['"{}"'.format(d) for d in IFOs])
+
+    def _get_setup_from_noiseSFTs(self):
+        noise_multi_sft_catalog = lalpulsar.GetMultiSFTCatalogView(
+            lalpulsar.SFTdataFind(self.noiseSFTs, self.SFTconstraint)
+        )
+
+        # Information to be extracted from the SFTs themselves
+        IFOs = []
+        tstart = []
+        tend = []
+        Tsft = []
+        self.sftfilenames = []  # This refers to the MFD output!
+
+        # SWIG-LAL does not import this functionality
+        gps_to_float = lambda x: x.gpsSeconds
+
+        for ifo_catalog in noise_multi_sft_catalog.data:
+            ifo_name = lalpulsar.ListIFOsInCatalog(ifo_catalog).data[0]
+
+            time_stamps = lalpulsar.TimestampsFromSFTCatalog(ifo_catalog)
+            this_start_time = gps_to_float(time_stamps.data[0])
+            this_end_time = gps_to_float(time_stamps.data[-1])
+            this_Tsft = int(round(1.0 / ifo_catalog.data[0].header.deltaF))
+
+            self.sftfilenames.append(
+                lalpulsar.OfficialSFTFilename(
+                    ifo_name[0],
+                    ifo_name[1],
+                    ifo_catalog.length,
+                    this_Tsft,
+                    this_start_time,
+                    this_end_time - this_start_time + this_Tsft,
+                    self.label,
+                )
+            )
+
+            IFOs.append(ifo_name)
+            tstart.append(this_start_time)
+            tend.append(this_end_time)
```

Comment:
```suggestion
            tend.append(this_end_time+ this_Tsft)
```
for consistency with the other case? (please double-check)